src/NHibernate/Dialect/FirebirdDialect.cs

```diff
@@ -431,7 +431,8 @@ private static bool IsUnallowedDecimal(DbType dbType, int precision)
 		/// Does this dialect support distributed transaction?
 		/// </summary>
 		/// <remarks>
-		/// As of v2.5, fails rollback-ing changes when distributed: changes are instead persisted in database.
+		/// As of v2.5 and 3.0.2, fails rollback-ing changes when distributed: changes are instead persisted in database.
+		/// (With ADO .Net Provider 5.9.1)
```

Comment:
Locally tested with Firebird 3.0.2, provider 5.9.1 (without using embedded mode), thanks to the auto-enlistment now done by default even at connection opening with the new option `AutoJoinTransaction`. The provider 2.5.2 used in Teamcity fails to connect, it is probably too old.

Just checked [their code](https://sourceforge.net/p/firebird/NETProvider/ci/master/tree/Provider/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbConnectionInternal.cs#l282):

```C#
if (_owningConnection != null && _options.Enlist)
``` 

Great, they use the auto-enlistment option as a global "enable system transaction" switch... While at the same time enabling it causes Firebird connection to fail opening if no ambient transaction are there...